RZDataBinding/NSObject+RZDataBinding.m

```diff
@@ -409,28 +388,62 @@ - (void)invalidate
 
 #pragma mark - RZDBObserverContainer implementation
 
-@implementation RZDBObserverContainer
+@implementation RZDBObserverContainer {
+    NSHashTable *_observers;
+}
 
-- (instancetype)init
++ (instancetype)strongContainer
+{
+    return [[self alloc] initWithBackingStore:[NSHashTable hashTableWithOptions:NSPointerFunctionsStrongMemory | NSPointerFunctionsObjectPersonality]];
+}
+
++ (instancetype)weakContainer
+{
+    return [[self alloc] initWithBackingStore:[NSHashTable weakObjectsHashTable]];
+}
+
+- (instancetype)initWithBackingStore:(NSHashTable *)backingStore
 {
     self = [super init];
     if ( self != nil ) {
-        _observers = [NSHashTable weakObjectsHashTable];
+        _observers = backingStore;
     }
     return self;
 }
 
 - (void)addObserver:(RZDBObserver *)observer
 {
     @synchronized (self) {
-        [self.observers addObject:observer];
+        [_observers addObject:observer];
     }
 }
 
 - (void)removeObserver:(RZDBObserver *)observer
 {
     @synchronized (self) {
-        [self.observers removeObject:observer];
+        [_observers removeObject:observer];
+    }
+}
+
+- (void)enumerateObserversUsingBlock:(void (^)(RZDBObserver *, BOOL *))block
+{
+    @synchronized (self) {
+        NSHashTable *observers = ^{
+            @synchronized (_observers) {
+                return [_observers copy];
+            }
+        }();
```

Comment:
What is this, swift?!

I was looking at replacing it with the C-based `({ })` format, but this still reads nicer.